rosbag2/src/rosbag2/readers/sequential_reader.cpp

```diff
@@ -108,4 +110,56 @@ std::vector<TopicMetadata> SequentialReader::get_all_topics_and_types()
   throw std::runtime_error("Bag is not open. Call open() before reading.");
 }
 
+bool SequentialReader::has_next_file() const
+{
+  return current_file_iterator_ + 1 != file_paths_.end();
+}
+
+void SequentialReader::load_next_file()
+{
+  assert(current_file_iterator_ != file_paths_.end());
+  current_file_iterator_++;
+}
+
+std::string SequentialReader::get_current_file() const
+{
+  return *current_file_iterator_;
+}
+
+std::string SequentialReader::get_current_uri() const
+{
+  auto current_file = get_current_file();
+  auto current_uri = rcpputils::fs::remove_extension(current_file);
```

Comment:
This will need to be updated if we have compression extensions (ex. *.db3.snappy)

Reply:
In fact this is current not being used. The current behavior of the sqlite plugin requires a full path to the database when opening for read-only.